src/TwigComponent/src/ComponentTemplateFinder.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Twig\Environment;
+
+/**
+ * @author Matheo Daninos <matheo.daninos@gmail.com>
+ */
+final class ComponentTemplateFinder implements ComponentTemplateFinderInterface
+{
+    public function __construct(
+        private Environment $environment
+    ) {
+    }
+
+    public function findAnonymousComponentTemplate(string $name): ?string
+    {
+        $loader = $this->environment->getLoader();
+        $componentPath = rtrim(str_replace(':', '/', $name));
+
+        if ($loader->exists($componentPath)) {
```

Comment:
Maybe a micro-optimization but should we order these checks to make what we call the best practice first? Here's my thinking:

1. `components/{$name}.html.twig`
2. `{$name}.html.twig`
3. `components/{$name}`
4. `{$name}`